config.py

```diff
@@ -705,9 +705,23 @@ def default_reject_comps_factory():
 # For example `export MNE_BIDS_STUDY_CONFIG=config_eeg_matchingpennies`
 
 if "MNE_BIDS_STUDY_CONFIG" in os.environ:
-    cfg_name = os.environ['MNE_BIDS_STUDY_CONFIG']
-    cfg_path = 'tests.configs.{}'.format(cfg_name)
-    custom_cfg = importlib.import_module(cfg_path)
+    cfg_path = os.environ['MNE_BIDS_STUDY_CONFIG']
+
+    if os.path.exists(cfg_path):
+        print('Using custom configuration specified in MNE_BIDS_STUDY_CONFIG.')
+    else:
+        msg = ('The custom configuration file specified in the '
+               'MNE_BIDS_STUDY_CONFIG environment variable could not be '
+               'found: {cfg_path}'.format(cfg_path=cfg_path))
+        raise ValueError(msg)
+
+    # Import configuration from an arbitrary path without having to fiddle
+    # with `sys.path`.
+    spec = importlib.util.spec_from_file_location(name='custom_config',
+                                                  location=cfg_path)
+    custom_cfg = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(custom_cfg)
+    del spec
```

Comment:
this is way too complicated to be user facing.

here's another thought. What if you had a starter script like sphinx which would prompt the user to enter the values in the `config.py` and then create a `config` script based on that?

Reply:
I also don't like the idea of forcing an end-user to set environment variables, but this here will make testing in my developing environments (with different datasets) much easier. I'm simply generalizing a functionality that's already been there which was only usable for CI tests so far.

Or do you mean: having this _code snippet_ inside `config.py` is too complicated for a user? I totally agree, but this applies to _many_ aspects of `config.py` IMHO. It's not user friendly. My personal preference would be to have a plaintext (YAML or INI) config file anyway, I don't like simply loading and executing arbitrary Python code. But this is far beyond the scope of this PR and should probably be discussed in the context of what I have mentioned in #83 – do we need a package etc. But for now, I just want to get stuff to work :)